src/super_gradients/training/metrics/detection_metrics.py

```diff
@@ -150,12 +179,22 @@ def compute(self) -> Dict[str, Union[float, torch.Tensor]]:
             # MaP is averaged over IoU thresholds and over classes
             mean_ap = ap.mean()
 
+            # Fill array of per-class AP scores with values for classes that were present in the dataset
+            ap_per_class = ap.mean(1)
+            for i, class_index in enumerate(unique_classes):
```

Comment:
Why not just mean_ap_per_class = ap.mean(1) ?

Reply:
The function that computes `ap` does not necessary returns all [num_classes]. I fact it returns `ap` as an array [Present Classes in Dataset, Num IoU thresholds].
In some unit tests (when we were using minicoco) it was the case when only 16 classes were returned instead of 80. 
And this is actually by design: https://github.com/Deci-AI/super-gradients/pull/1238/files#diff-704b8d5cb142a02bea1be07634201556203fe6f2a0d17e5ad6216b73c0f07fbbR1130
So that's why we have this filling procedure here.

Reply:
mean_ap_per_class[unique_classes] = ap_per_class looks nicer to me (And probably a tiny bit more efficient), but might be less readable. Your call.